.github/workflows/publish-draft-releases.yml

```diff
@@ -1132,41 +1140,49 @@ jobs:
         with:
           ruby-version: 2.7
       -
-        name: get runtime version
-        id: get-runtime-version-current
+        name: get ${{ matrix.runtime.name }} runtime version current
+        id: get-runtime-current
         run: |
-          runtime_ver_current="$(ruby -e '
+          runtime_current="$(ruby -e '
             require "./scripts/github/lib.rb";
-            puts get_runtime("calamari")
+            puts get_runtime("${{ matrix.runtime.name }}")
           ')"
-          echo "::set-output name=runtime-ver-current::$runtime_ver_current"
-          echo $runtime_ver_current
+          echo "::set-output name=${{ matrix.runtime.name }}-runtime-current::$runtime_current"
           echo ....................
-      - name: Show new runtime version
-        run: echo "The selected value is ${{ steps.get-runtime-version-current.outputs.runtime-ver-current }}"
       -
-        name: get base runtime version
-        id: get-runtime-version-base
+        name: get ${{ matrix.runtime.name }} runtime version base
+        id: get-runtime-base
         run: |
           mkdir temp_for_run
           cd temp_for_run
           git clone -b manta https://github.com/Manta-Network/Manta.git
           cd Manta
-          runtime_ver_base="$(ruby -e '
+          runtime_base="$(ruby -e '
             require "./scripts/github/lib.rb";
-            puts get_runtime("calamari")
+            puts get_runtime("${{ matrix.runtime.name }}")
           ')"
-          echo "::set-output name=runtime-ver-base::$runtime_ver_base"
-          echo $runtime_ver_base
+          echo "::set-output name=${{ matrix.runtime.name }}-runtime-base::$runtime_base"
           echo ....................
-      - name: Show base runtime version
-        run: echo "The selected value is ${{ steps.get-runtime-version-base.outputs.runtime-ver-base }}"
+
+  check-for-runtime-upgrade:
+    needs:
+      - parse-runtimes
+    runs-on: ubuntu-20.04
+    outputs:
+      do-versions-match: ${{ steps.check-match.outputs.match }}
+    steps:
       -
-        name: check-match
+        name: check if runtime versions match
         id: check-match
         run: |
-          echo "::set-output name=match::false"
-          if [[ ${{ steps.get-runtime-version-base.outputs.runtime-ver-base }} == ${{ steps.get-runtime-version-current.outputs.runtime-ver-current }} ]]; then echo "::set-output name=match::true"
-          fi
+          echo "::set-output name=match::true"
+          if [[ ${{ needs.parse-runtimes.outputs.manta-runtime-base }} != ${{ needs.parse-runtimes.outputs.manta-runtime-current }} ]]; then echo "::set-output name=match::false"; fi
+          if [[ ${{ needs.parse-runtimes.outputs.calamari-runtime-base }} != ${{ needs.parse-runtimes.outputs.calamari-runtime-current }} ]]; then echo "::set-output name=match::false"; fi
```

Comment:
We've parsed 4 runtime versions. 2 old and 2 new.
The logic is that if either calamari's runtimes don't match or manta's runtime don't match the test will be triggered.

Reply:
Can we fail the test in addition to the echo?

Reply:
Yep can do.